models/models.go

```diff
@@ -65,10 +73,33 @@ type StageWorkflowModel struct {
 	RunIf string `json:"run_if,omitempty" yaml:"run_if,omitempty"`
 }
 
-type DagWorkflowListItemModel map[string]DagWorkflowModel
+type GraphPipelineWorkflowListItemModel map[string]GraphPipelineWorkflowModel
+
+type GraphPipelineWorkflowModel struct {
+	DependsOn       []string                   `json:"depends_on,omitempty" yaml:"depends_on,omitempty"`
+	AbortOnFail     bool                       `json:"abort_on_fail,omitempty" yaml:"abort_on_fail,omitempty"`
+	RunIf           GraphPipelineRunIfModel    `json:"run_if,omitempty" yaml:"run_if,omitempty"`
+	ShouldAlwaysRun GraphPipelineAlwaysRunMode `json:"should_always_run,omitempty" yaml:"should_always_run,omitempty"`
+}
+
+type GraphPipelineRunIfModel struct {
+	Expression string `json:"expression,omitempty" yaml:"expression,omitempty"`
+}
+
+func (d *GraphPipelineAlwaysRunMode) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var value string
+	if err := unmarshal(&value); err != nil {
+		return err
+	}
+
+	allowedValues := []string{string(GraphPipelineAlwaysRunModeOff), string(GraphPipelineAlwaysRunModeWorkflow)}
+	if !slices.Contains(allowedValues, value) {
+		return fmt.Errorf("%s is not a valid should always run value", value)
```

Comment:
```suggestion
		return fmt.Errorf("%s is not a valid should_always_run value", value)
```

Reply:
Maybe also print the allowed values too? Given that it's already there in a slice.

Reply:
Good ideas. I have updated the code.